### PR TITLE
[front] - chore: enforce at least one provider

### LINKED ISSUE #5505: [Model Deactivation] - enforce at least one provider

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at ensuring that an admin selects at least one model provider.

## Risk

None as this component is not being used yet.

## Deploy Plan

Not to be deployed.

### CODE REVIEW

File: front/components/workspace/ProviderManagementModal.tsx [hunk 1]
```diff
@@ -65,19 +67,31 @@ export function ProviderManagementModal({
     },
     [setProviderStates]
   );
+
+  const handleSave = useCallback(() => {
+    const activeProviders = MODEL_PROVIDER_IDS.filter(
+      (key) => providerStates[key]
+    );
```
Comment (JulesBelveze) [inline front/components/workspace/ProviderManagementModal.tsx:74]: Not sure if the following is actually preferred ?
```typescript
const providerStatesKeys = Object.keys(providerStates) as ModelProviderIdType[];
const activeProviders = providerStatesKeys.filter(
   (key) => providerStates[key]
);
```

File: front/components/workspace/ProviderManagementModal.tsx [hunk 2]
```diff
@@ -65,19 +67,31 @@ export function ProviderManagementModal({
     },
     [setProviderStates]
   );
+
+  const handleSave = useCallback(() => {
+    const activeProviders = MODEL_PROVIDER_IDS.filter(
```
Comment (spolu) [inline front/components/workspace/ProviderManagementModal.tsx:72]: Why use a useCallback?

File: front/components/workspace/ProviderManagementModal.tsx (hunk 1 above)
Comment (spolu) [reply front/components/workspace/ProviderManagementModal.tsx:74]: Use of as is to be avoided as much as possible 👍

**Comment by spolu** [comment]
Is this also enforced server side ?

File: front/components/workspace/ProviderManagementModal.tsx (hunk 2 above)
Comment (JulesBelveze) [reply front/components/workspace/ProviderManagementModal.tsx:72]: True, function is quite straightforward and probably wouldn't significantly benefit from memoization. Will ditch it

**Review by tdraier — APPROVED** [review]

**Pull request merged**